include/mbgl/shaders/vulkan/common.hpp

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+#include <mbgl/shaders/shader_source.hpp>
+
+namespace mbgl {
+namespace shaders {
+
+template <>
+struct ShaderSource<BuiltIn::Prelude, gfx::Backend::Type::Vulkan> {
+    static constexpr const char* name = "Prelude";
+
+    static constexpr auto vertex = R"(
+
+#define M_PI 3.1415926535897932384626433832795
+
+// The maximum allowed miter limit is 2.0 at the moment. the extrude normal is stored
+// in a byte (-128..127). We scale regular normals up to length 63, but there are also
+// "special" normals that have a bigger length (of up to 126 in this case).
+#define LINE_NORMAL_SCALE (1.0 / (127 / 2))
+
+// The attribute conveying progress along a line is scaled to [0, 2^15).
+#define MAX_LINE_DISTANCE 32767.0
+
+// Unpack a pair of values that have been packed into a single float.
```

Comment:
This will now the third instance of these comment in the repo. Is this just one of those things you don't apply DRY to? Or do we have any other option here?

Reply:
We currently have one shader predicate for GL and one for Metal, since they have different syntax.  But since they both understand `#define`, we could have a pre-predicate that's shared between all three backends, and perhaps another with stuff that's shared just between GL and Vulkan.